engine/pkg/election/elector.go

```diff
@@ -247,31 +220,55 @@ func (e *Elector) release(ctx context.Context, removeSelf bool) error {
 	ctx, cancel := context.WithTimeout(ctx, defaultReleaseTimeout)
 	defer cancel()
 
-	s := e.config.Storage
+	return e.updateRecord(ctx, func(record *Record) error {
+		if record.LeaderID == e.config.ID {
+			record.LeaderID = ""
+		}
+		if removeSelf {
+			for i, m := range record.Members {
+				if m.ID == e.config.ID {
+					record.Members = append(record.Members[:i], record.Members[i+1:]...)
+					break
+				}
+			}
+		}
+		return nil
+	})
+}
 
-	record, err := s.Get(ctx)
-	if err != nil {
-		return errors.Trace(err)
+func (e *Elector) updateRecord(ctx context.Context, f func(*Record) error) error {
+	// Divide 2 is for more retries.
+	backoffBaseDelayInMs := int64(e.config.RenewInterval/time.Millisecond) / 2
+	// Make sure the retry delay is less than the deadline, otherwise the retry has no chance to execute.
+	backoffMaxDelayInMs := int64(e.config.RenewDeadline/time.Millisecond) / 2
+	if deadline, ok := ctx.Deadline(); ok {
+		backoffMaxDelayInMs = int64(deadline.Sub(time.Now())/time.Millisecond) / 2
```

Comment:
Should use the smaller one for `backoffMaxDelayInMs`

Reply:
Addressed in https://github.com/pingcap/tiflow/pull/7338/commits/58fa73cdf26b71098210e80fe8d4212227b53cd4.